### DISCUSSION TITLE
Implementing a custom pyrolysis model

### DISCUSSION THREAD

**Discussion by r-broek** [firemodels/fds#13069] [Q&A] [answered]
Dear developers, 

I am trying to implement a custom pyrolysis model into the source code, which I would like to turn on using the argument `CONVECTION_MODEL = T`. I have modified `wall.f90` to include an extra `CASE` and defined this in `cons.f90` as follows:
`INTEGER, PARAMETER :: PYROLYSIS_CONVECTION_MODEL=6`.

In `read.f90` I added the `LOGICAL` `CONVECTION_MODEL`, with its default value set to `.FALSE.`. Furthermore, this argument was also appended to the end of the `namelist /MATL/`.

Lastly, the if statement
```
IF (N_REACTIONS > 0 .AND. CONVECTION_MODEL) THEN
      WRITE(MESSAGE,'(A,A,A)') 'ERROR(256): MATL ',TRIM(ID),', CONVECTION_MODEL applied.'
ENDIF
```
was used to see whether the model had been turned on. 
Unfortunately, I have missed something, as the 'CONVECTION_MODEL applied.' message was not output when I ran a simulation. 

Hopefully, someone can help me out with what I have missed.

**Comment by drjfloyd** [comment, marked as answer]
You should make sure you understand what all variables are when modifying FDS source. MESSAGE is a CHARACTER variable and not a logical unit. So writing data to MESSAGE is just storing data in a variable.  Look at how all the WARNING messages are written for how you need to do this.

**Reply by r-broek** [reply]
Oh I see. Sorry for being oblivious, and thank you for your quick reply.
